test/clj/chronograph/domain/timer_test.clj

```diff
@@ -0,0 +1,480 @@
+(use-fixtures :each
+  fixtures/clear-db
+  setup-members-org-tasks-fixture)
+
+
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Create Timer Tests
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(deftest create-timer-test
+  (let [task-id (:task-id-1 @test-context)
+        member-id (:member-id @test-context)]
+    (testing "Timer creation"
+      (is (s/valid? :timers/timer
+                    (timer/create! db/datasource
+                                   member-id
+                                   task-id
+                                   nil))
+          "without a note, returns a valid timer.")
+      (is (s/valid? :timers/timer
+                    (timer/create! db/datasource
+                                   member-id
+                                   task-id
+                                   "A valid note."))
+          "with a note, returns a valid timer.")
+      (is (every? (partial s/valid? :timers/timer)
+                  (into (repeatedly 3 (fn [] (timer/create! db/datasource
+                                                            member-id
+                                                            task-id
+                                                            "A sample note.")))
+                        (repeatedly 3 (fn [] (timer/create! db/datasource
+                                                            member-id
+                                                            (:task-id-2 @test-context)
+                                                            "A sample note.")))))
+          "lets a member create multiple timers for multiple tasks.")
+      (let [timer (timer/create! db/datasource
+                                 member-id
+                                 task-id
+                                 "A sample note.")]
+        (is (not (timer/running? db/datasource
+                                 member-id
+                                 (:timers/id timer)))
+            "does not implicitly start the timer."))
+      (is (thrown-with-msg? PSQLException
+                            #"ERROR: insert or update on table \"timers\" violates foreign key constraint \"timers_user_id_fkey\""
+                            (timer/create! db/datasource
+                                           Long/MAX_VALUE
+                                           task-id
+                                           "A sample note."))
+          "with a non-existing user fails with a PG exception.")
+      (is (thrown-with-msg? PSQLException
+                            #"ERROR: insert or update on table \"timers\" violates foreign key constraint \"timers_task_id_fkey\""
+                            (timer/create! db/datasource
+                                           member-id
+                                           Long/MAX_VALUE
+                                           "A sample note."))
+          "with a non-existing task fails with a PG exception."))))
```

Comment:
Check that the created timer actually exists in the DB.

Reply:
Fixed, however I'm wondering if it is necessary, given that we say `:return-keys true`. 

next-jdbc will return the keys IFF the write succeeds.